integrations/fastembed/src/haystack_integrations/components/embedders/fastembed/embedding_backend/fastembed_backend.py

```diff
@@ -39,7 +41,12 @@ def __init__(
     ):
         self.model = TextEmbedding(model_name=model_name, cache_dir=cache_dir, threads=threads)
 
-    def embed(self, data: List[List[str]], **kwargs) -> List[List[float]]:
+    def embed(self, data: List[str], progress_bar=True, **kwargs) -> List[List[float]]:
         # the embed method returns a Iterable[np.ndarray], so we convert it to a list of lists
-        embeddings = [np_array.tolist() for np_array in self.model.embed(data, **kwargs)]
+        embeddings = []
+        embeddings_iterable = self.model.embed(data, **kwargs)
+        for np_array in tqdm(
+            embeddings_iterable, disable=not progress_bar, desc="Calculating embeddings", total=len(data)
+        ):
+            embeddings.append(np_array.tolist())
```

Comment:
since the original library does not provide this feature,
we create the progress bar using `tqdm` in the embedding backend.